src/commands/mkdir.rs

```diff
@@ -0,0 +1,63 @@
+use crate::errors::ShellError;
+use crate::parser::hir::SyntaxType;
+use crate::parser::registry::{CommandConfig, NamedType, PositionalType};
+use crate::prelude::*;
+use indexmap::IndexMap;
+use std::path::{Path, PathBuf};
+
+pub struct Mkdir;
+
+impl Command for Mkdir {
+    fn run(&self, args: CommandArgs) -> Result<OutputStream, ShellError> {
+        mkdir(args)
+    }
+
+    fn name(&self) -> &str {
+        "mkdir"
+    }
+
+    fn config(&self) -> CommandConfig {
+        let mut named: IndexMap<String, NamedType> = IndexMap::new();
+        named.insert("p".to_string(), NamedType::Switch);
+
+        CommandConfig {
+            name: self.name().to_string(),
+            positional: vec![PositionalType::mandatory("file", SyntaxType::Path)],
+            rest_positional: false,
+            named,
+            is_sink: false,
+            is_filter: false,
+        }
+    }
+}
+
+pub fn mkdir(args: CommandArgs) -> Result<OutputStream, ShellError> {
+    let env = args.env.lock().unwrap();
+    let path = env.path.to_path_buf();
+    let mut full_path = PathBuf::from(path);
+
+    match &args.nth(0) {
+        Some(Tagged { item: value, .. }) => full_path.push(Path::new(&value.as_string()?)),
+        _ => {}
+    }
+
+    if !args.has("p") {
```

Comment:
Let's use the full name of the switch, first. We'll have short-name switches in the future we can use to alias them.

Reply:
I was not sure what to name the long form flag. What name do you suggest?

Reply:
I named it `--create-all`

Reply:
nixed `--create-all` and discarded alternative `--deep` by making it the default as discussed.